File: basex/jython/read/basex_read.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	basex_read.py
#
#					May/31/2012
#
# ------------------------------------------------------------------
import	sys
import	httplib
import	xml.dom.minidom
from java.lang import	String
from java.lang import	System
#
sys.path.append ("/var/www/data_base/common/python_common")
sys.path.append ("/var/www/data_base/common/jython_common")
#
from jython_xml_manipulate import jython_xml_to_dict_proc
from text_manipulate import dict_display_proc
#
# ------------------------------------------------------------------
System.out.println (String ("*** 開始 ***","utf-8"))
#
#
conn = httplib.HTTPConnection ("localhost:8984")
conn.request ("GET","/rest/cities?query=/")
r1 = conn.getresponse ()
print r1.status, r1.reason
xml_str = r1.read ()
conn.close ()
#
dict_aa = jython_xml_to_dict_proc (xml_str)
dict_display_proc (dict_aa)
#
System.out.println (String ("*** 終了 ***","utf-8"))
# ------------------------------------------------------------------

File: basex/ruby/read/basex_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#	basex_read.rb
#
#						Oct/27/2011
#
# ------------------------------------------------------------
require "rexml/document"
require "restclient"
#
load '/var/www/data_base/common/ruby_common/xml_manipulate.rb'
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
include REXML
#
# ------------------------------------------------------------
puts	"*** 開始 ***"
#
url_in="http://admin:admin@localhost:8984/rest/cities?query=/"
#
xml_str = RestClient.get url_in
dict_aa=xml_to_dict_proc(xml_str)
dict_display_proc(dict_aa)
#
puts "*** 終了 ***"
# ------------------------------------------------------------
